tortoisebeacon/weakcoin/weak_coin_test.go

```diff
@@ -1,57 +1,118 @@
-//	wcg := NewWeakCoin(DefaultPrefix, DefaultThreshold, nil, logger)
-//	epoch := types.EpochID(3)
-//	round := types.RoundID(1)
-//	expected := "0xb150cef2e30473bf3b1d24e253cce39f995c15ebc37bb019d7ab618e873509fc"
-//
-//	p, err := wcg.(*weakCoin).generateProposal(epoch, round)
-//	r.NoError(err)
-//
-//	r.EqualValues(expected, p.String())
-//}
-
-//func Test_weakCoin_proposalExceedsThreshold(t *testing.T) {
-//	t.Parallel()
-//
-//	r := require.New(t)
-//
-//	logger := log.NewDefault("WeakCoin")
-//
-//	tt := []struct {
-//		name      string
-//		proposal  types.Hash32
-//		threshold types.Hash32
-//		expected  bool
-//	}{
-//		{
-//			name:      "Case 1",
-//			proposal:  types.HexToHash32("0xb150cef2e30473bf3b1d24e253cce39f995c15ebc37bb019d7ab618e873509fc"),
-//			threshold: types.HexToHash32("0x80" + strings.Repeat("00", 31)),
-//			expected:  true,
-//		},
-//		{
-//			name:      "Case 2",
-//			proposal:  types.HexToHash32("0x01"),
-//			threshold: types.HexToHash32("0x80" + strings.Repeat("00", 31)),
-//			expected:  false,
-//		},
-//	}
-//
-//	for _, tc := range tt {
-//		tc := tc
-//		t.Run(tc.name, func(t *testing.T) {
-//			t.Parallel()
-//
-//			wcg := NewWeakCoin(DefaultPrefix, DefaultThreshold, nil, logger)
-//			got := wcg.(*weakCoin).proposalExceedsThreshold(tc.proposal)
-//			r.Equal(tc.expected, got)
-//		})
-//	}
-//}
+import (
+	"encoding/hex"
+	"math/big"
+	"testing"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_weakCoin_generateProposal(t *testing.T) {
+	t.Parallel()
+
+	r := require.New(t)
+
+	var wc weakCoin
+	epoch := types.EpochID(3)
+	round := types.RoundID(1)
+	expected := "5765616b436f696e03000000000000000100000000000000"
+
+	p, err := wc.generateProposal(epoch, round)
+	r.NoError(err)
+
+	r.EqualValues(expected, hex.EncodeToString(p))
+}
+
+func Test_weakCoin_proposalExceedsThreshold(t *testing.T) {
+	t.Parallel()
+
+	r := require.New(t)
+
+	proposal1, ok := new(big.Int).SetString("0xb150cef2e30473bf3b1d24e253cce39f995c15ebc37bb019d7ab618e873509fc", 0)
+	require.True(t, ok)
+
+	proposal2, ok := new(big.Int).SetString("0x01", 0)
+	require.True(t, ok)
+
+	threshold, ok := new(big.Int).SetString("0x8000000000000000000000000000000000000000000000000000000000000000", 0)
+	require.True(t, ok)
+
+	tt := []struct {
+		name      string
+		proposal  []byte
+		threshold []byte
+		expected  bool
+	}{
+		{
+			name:      "Case 1",
+			proposal:  proposal1.Bytes(),
+			threshold: threshold.Bytes(),
+			expected:  true,
+		},
+		{
+			name:      "Case 2",
+			proposal:  proposal2.Bytes(),
+			threshold: threshold.Bytes(),
+			expected:  false,
+		},
+	}
+
+	for _, tc := range tt {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			var wc weakCoin
+			wc.threshold = DefaultThreshold()
+			got := wc.proposalExceedsThreshold(tc.proposal)
+			r.Equal(tc.expected, got)
+		})
+	}
+}
+
+func Test_weakCoin_coinBitToBool(t *testing.T) {
```

Comment:
i would remove this utility completely. bit can't be any other value from 0 or 1

Reply:
Well, I added it to just indicate that calculations are correct. With `Bit(0)` it's not needed because `Bit` is tested in the standard library